format/spec.md

```diff
@@ -648,6 +648,21 @@ Notes:
 5. The `content_offset` and `content_size_in_bytes` fields are used to reference a specific blob for direct access to a deletion vector. For deletion vectors, these values are required and must exactly match the `offset` and `length` stored in the Puffin footer for the deletion vector blob.
 6. The following field ids are reserved on `data_file`: 141.
 
+For Variant, values in the `lower_bounds` and `upper_bounds` maps store the a serialized Variant object that contains lower and upper bounds for fields within the Variant. The object keys are normalized JSON path expressions that uniquely identify a Variant field. The object values are primitive Variant representations of the lower or upper bound for the field. Including bounds for any field is optional and the bounds must have the same Variant type.
+
+Bounds for a field must be accurate for all non-null values of the field in a data file. Bounds for values within arrays must be accurate all values in the array. Bounds must not be written to describe values with mixed Variant types (other than null). For example, a "measurement" field that contains int64 and null values may have bounds, but a string value such as "n/a" or "0" in any record would cause the bounds to be skipped.
```

Comment:
```suggestion
Bounds for a field must be accurate for all non-null values of the field in a data file. Bounds for values within arrays must be accurate all values in the array. Bounds must not be written to describe values with mixed Variant types (other than null). For example, a "measurement" field that contains int64 and null values may have bounds, but if the field also contained a  string value such as "n/a" or "0" then bounds may not be stored.
```

Reply:
I want to be sure, but the last sentence should be a rule correct? You may not store bounds if multiple types exist for the field

Reply:
That's correct. The problem with "0" is that it assumes engine behavior if we either cast it to 0 or if we don't. We want to match the engine's definition of when a row matches, which means being conservative and leaving casting an engine responsibility.